lib/src/components/NxProgressBar/NxProgressBar.scss

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @use "sass:math";
+
+.nx-progress-bar {
+  display: flex;
+  flex-direction: column;
+  gap: var(--nx-spacing-2x);
+  margin: var(--nx-spacing-6x) 0;
+
+  &--inline-counter {
+    flex-direction: row;
+  }
+
+  &--inline {
+    display: inline-flex;
+    margin: 0 var(--nx-spacing-6x);
+    vertical-align: middle;
+  }
+
+  &--success {
+    color: var(--nx-swatch-green-25);
+  
+    .nx-progress-bar__counter {
+      border-color: var(--nx-swatch-green-25);
+      background-color: var(--nx-swatch-green-95);
+    }
+  }
+  
+  &--error {
+    color: var(--nx-swatch-red-40);
+  
+    .nx-progress-bar__counter {
+      border-color: var(--nx-swatch-red-40);
+      background-color: var(--nx-swatch-red-95);
+    }
+  }
+}
+
+.nx-progress-bar__counter-and-label {
+  .nx-icon {
+    margin: 0 var(--nx-spacing-2x);
+  }
+}
+
+.nx-progress-bar__counter {
+  margin: 0 var(--nx-spacing-2x) 0 0;
```

Comment:
The way these margins are set up causes a minor problem: when in inline-counter mode, the counter's right margin is still active, effectively mandating that much extra space at the right end of the component. This is particularly relevant in full-width mode, where that extra space makes it not-quite-full-width.

My recommendation would be for each of the three second-row elements - the counter, icon, and label - to exclusively use left margin to manage the expected space between them.

Reply:
> The way these margins are set up causes a minor problem: when in inline-counter mode, the counter's right margin is still active, effectively mandating that much extra space at the right end of the component. This is particularly relevant in full-width mode, where that extra space makes it not-quite-full-width.
> 
> My recommendation would be for each of the three second-row elements - the counter, icon, and label - to exclusively use left margin to manage the expected space between them.

I changed the margins for counter, icon, and label to left margin but I noticed that the spacing between the counter and the icon or label is larger. I thought it might be more difficult to manage that extra spacing with left margin because there are cases where it might be icon + label or just the label itself.

I added the right margin back here but removed it when it is inline-counter in these two commits.
(Also added vertical-align middle to center vertically)
Let me know your thoughts:

cbc48b5d5441963ab729607b574a50e4a57848d6
66f39ea76b30c03c5ba9f4630401f0fc3f0f74ce